src/sardana/macroserver/test/res/macros/testmacros.py

```diff
@@ -0,0 +1,148 @@
+
+        expect_params = (99, [1., 2.])
+        macro, pars = self.createMacro('pt6_base', *expect_params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (99, 1., 2.)
+        expect_params = (99, [1., 2.])
+        self.runMacro(macro)
+        macro, pars = self.createMacro('pt6_base', *params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([92], True)
+        macro, _ = self.createMacro("pt10_base", *expect_params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (91, True)
+        expect_params = ([91.], True)
+        macro, _ = self.createMacro("pt10_base", *params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+
+class execMacro(Macro):
+
+    def run(self, *args):
+
+        expect_params = (99, [1., 2.])
+        macro = self.execMacro('pt6_base', *expect_params)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (99, 1., 2.)
+        macro = self.execMacro('pt6_base', *params)
+        result = macro.data
+        expect_params = (99, [1., 2.])
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([92], True)
+        macro = self.execMacro('pt10_base', *expect_params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([99.], True)
+        macro = self.execMacro("pt10_base [99] True")
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([999.], True)
+        macro = self.execMacro("pt10_base", 999, True)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+
+class pt6_base(Macro):
+    """Macro with a number parameter followed by a list of numbers.
+    Usages from Spock, ex.:
+    pt6_base 99 [1 3]
+    pt6_base 99 1 3
+    """
+
+    param_def = [
+        ['val1', Type.Float, None, 'value 1'],
+        ['numb_list', [['pos', Type.Float, None, 'value']], None,
+         'List of values'],
+    ]
+
+    def run(self, *args, **kwargs):
+        self.data = args
+
+
+class pt10_base(Macro):
+    """Macro with a list of numbers followed by a number parameter.
+    Usages from Spock, ex.:
+    pt10_base 99 [1 3]
```

Comment:
I think this is not correct, the first parameters is repeat parameter and not the second one.

Reply:
Solved at [commit 6923430](https://github.com/sardana-org/sardana/pull/781/commits/6923430d779d98e034b2f19930d42d0e6d6e5185)